src/internal/collection/types.go

```diff
@@ -42,6 +42,8 @@ type EtcdCollection interface {
 	// Claim attempts to claim a key and run the passed in callback with
 	// the context for the claim.
 	Claim(ctx context.Context, key string, val proto.Message, f func(context.Context) error) error
+
+	WithRenewer(ctx context.Context, cb func(context.Context, *Renewer) error) error
```

Comment:
The callback is necessary here so that a new context scoped to the renewer can be passed through?

Reply:
Yes.